### DISCUSSION TITLE
Workspace interactive shell or workspace browser

### DISCUSSION THREAD

**Discussion by fernap3** [community/community#25145] [Actions]
<p>A common workflow I use when debugging builds using other build tools is to inspect the workspace after a build has been completed.  Sometimes it’s super useful to know what files are where, etc., and currently the only way I can see to do that in Actions is to alter my action to print out a directory listing.  But that makes the develop/test loop for the build process quite long as I have to wait for another build to complete to get feedback.  And I have no real way of printing out directory info if the build fails.</p>
<p>Some other tools that provide solutions for this:</p>
<ul>
<li>Jenkins has a workspace viewer, allowing you to see the filesystem after a build completes</li>
<li>Heroku allows you shell access to the app instance which you can effectively use to inspect the environment</li>
</ul>
<p>Is it posssible to get a solution to this?</p>
<p>Also, does anyone know if there is documentation as to the state of the workspace between builds?  Is a fresh VM used for each build?  Is the workspace cleaned between builds?</p>
<p>Any advice for build debugging in general would be appreciated.</p>
<p>Thanks!<br>
Peter</p>

**Comment by csexton** [comment]
<p>I put something together to help me debug Xcode build last week:</p>
<p><a href="https://github.com/csexton/action-debugger" rel="nofollow noopener">https://github.com/csexton/action-debugger</a></p>
<p>Supports linux and mac and can give you a ssh command to connect to the action and poke around. Should be able to use it by dropping <code>&amp;nbsp;uses: csexton/action-debugger@v1</code> as a step for your build.</p>

**Comment by mpdude** [comment, marked as answer]
<p><a href="https://github.com/mxschmitt/action-tmate" rel="nofollow noopener">https://github.com/mxschmitt/action-tmate</a></p>
<p>This allows you to SSH into the worker node.</p>

**Comment by eMPee584** [comment]
<p>uber essential tool ^^</p>
<p>oh my had i discovered this yesterday i wouldn’t have created a franken branch with erratic commits to bump the build bot and only be able to debug from self logged state in the VM…<br>
live debugging is so much more convenient :sunglasses:</p>
